From e9ce1fa006b7e409b889fc7af9105e629b00bbf6 Mon Sep 17 00:00:00 2001
From: Jeremy Rocher <jeremy.rocher@intel.com>
Date: Wed, 1 Jul 2015 09:16:36 +0200
Subject: [PATCH 3/3] pstore: remove id on console and ftrace entries to comply
 with Android userspace

Since a recent kernel change [1] an id is appended to pstore console-*
and ftrace-* files. This cause issues as some components of Android
userspace relies on this file name without the id : system_server,
dumpstate, healthd.

To "fix" this, Google in its last patch on pstore removed the id
[2]. This patch, and its variants, was cherry-picked on several
kernels internally (gmin, cht) but not on gmin-dev as it comes from
upstream [3]. As the upstream version do not remove the id, this
explains the issue.

Looking at AOSP userspace code, it doesn't contain any change to
take this id in account.

Proposed fix : remove this id in kernel gmin-dev to be aligned with
other kernels and userspace.

1 - http://thread.gmane.org/gmane.linux.kernel.commits.head/488048
2 - https://android-review.googlesource.com/#/c/128956/
3 - http://thread.gmane.org/gmane.linux.kernel/1865449

Change-Id: I6e0a1d55d36f869e9506b89e231cb6bbfa6dc2c4
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-22916
Signed-off-by: Jeremy Rocher <jeremy.rocher@intel.com>
Signed-off-by: Imougar, JacquesX <jaquesx.imougar@intel.com>
Reviewed-on: https://android.intel.com:443/478321
---
 fs/pstore/inode.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/fs/pstore/inode.c b/fs/pstore/inode.c
index a16b90c..c6e2faa 100644
--- a/fs/pstore/inode.c
+++ b/fs/pstore/inode.c
@@ -319,10 +319,10 @@ int pstore_mkfile(enum pstore_type_id type, char *psname, u64 id, int count,
 			  psname, id, compressed ? ".enc.z" : "");
 		break;
 	case PSTORE_TYPE_CONSOLE:
-		scnprintf(name, sizeof(name), "console-%s-%lld", psname, id);
+		scnprintf(name, sizeof(name), "console-%s", psname);
 		break;
 	case PSTORE_TYPE_FTRACE:
-		scnprintf(name, sizeof(name), "ftrace-%s-%lld", psname, id);
+		scnprintf(name, sizeof(name), "ftrace-%s", psname);
 		break;
 	case PSTORE_TYPE_MCE:
 		scnprintf(name, sizeof(name), "mce-%s-%lld", psname, id);
-- 
1.9.1

